drivers/serial/Kconfig.stm32

```
@@ -59,6 +59,7 @@ config UART_STM32U5_ERRATA_DMAT_LOWPOWER
 
 config UART_STM32U5_ERRATA_DMAT_NOCLEAR
 	bool "Do not clear DMAT"
+	depends on !POWEROFF
```

Non-blocking: Only occurs to me now but we should comment why this exists e.g.,
```suggestion
	depends on !POWEROFF  # DMAT must be clear to enter Shutdown mode
```
(s/Shutdown/whatever the LP mode name is)